base/ltcmd.dtx

```diff
@@ -3506,79 +3506,79 @@
 % Some messages intended as errors when defining commands/environments.
 %    \begin{macrocode}
 \msg_new:nnnn { cmd } { arg-after-body }
-  { In~the~definition~of~#1,~b~(body)~argument~must~be~last. }
+  { Invalid~argument~type~'#2'~in~#1. }
```

Comment:
Part of the reasons for rewrites is to keep error messages to a single line and explanations to multiple lines, the other is to standardizes on less error messages.
It is more important that a grep shows the problematic csname #1 than the part of the more detailed explanation.
I don't see that the message  as given is wrong the argument  type 'm' is invalid whether or not that is because it  is always invalid or invalid in that position.

Reply:
> This also removes this particular message from the cmd namespace, so redirection, if used, won't affect this one message. I don't think this improves things. Rather, if this is only used by xparse it should be moved.

It could get moved but that wouldn't change its behavior, would it? And as you say, it would need some arkward syncing without actual gain (as far as I can see) which is why I just changed the module. I don't mind keeping it "cmd" but it is a package error and for some reason the code was left in the kernel but it gets never executed unless the package is loaded.

Reply:
ok, how about `{ Argument~type~'b'~must~be~last~in~#1. }` ?

Reply:
Sounds good too.  I didn't remove `Invalid` because I thought you wanted that as the first word for sorting.

Reply:
only wanted the "Invalid" if the messages  are then identical, if not different odering (in line with other messages) is fine as long as it is short :-)